File: src/main/kotlin/at/searles/parsingtools/common/SyntaxInfo.java
package at.searles.parsingtools.common;

import at.searles.parsing.ParserStream;

public class SyntaxInfo {

    private final long start;
    private final long end;

    protected SyntaxInfo(ParserStream stream) {
        this.start = stream.getStart();
        this.end = stream.getEnd();
    }

    protected SyntaxInfo(SyntaxInfo node) {
        this.start = node.getStart();
        this.end = node.getEnd();
    }

    public String toString() {
        return String.format("[%d:%d]", getStart(), getEnd());
    }

    public long getStart() {
        return start;
    }

    public long getEnd() {
        return end;
    }
}

File: src/main/kotlin/at/searles/parsingtools/common/SyntaxInfo.kt
package at.searles.parsingtools.common

import at.searles.parsing.ParserStream

open class SyntaxInfo {

    val start: Long
    val end: Long

    protected constructor(stream: ParserStream) {
        this.start = stream.start
        this.end = stream.end
    }

    protected constructor(node: SyntaxInfo) {
        this.start = node.start
        this.end = node.end
    }

    override fun toString(): String {
        return String.format("[%d:%d]", start, end)
    }
}
